.changelog/unreleased/improvements/581-remove-event-function-type-from-indexer.md

```diff
@@ -0,0 +1,2 @@
+- `[kvindex]` Remove event function type from event key in kvindexer. The function type is not always `FinalizeBlock`.
+  ([\#581](https://github.com/cometbft/cometbft/pull/581))
```

Comment:
What's the impact here on the user? If there's no direct impact on the user, it's not clear to me that we'd need a changelog entry. If the overall goal is to reduce storage consumption, then I'd recommend rewording to something that explicitly highlights that. Users may not understand what this changelog entry means for them otherwise.

Also, if it's being backported to v0.38, then the PR to `main` shouldn't have a changelog entry - only the backport PR should have a changelog entry (from a user's perspective, they'd be receiving this change in the v0.38 release series, so there'd be no need to also be notified about it in the v0.39 series).

Reply:
Makes sense. I will remove the changelog entry here. As for the impact on the user, there is no direct impact as far as I can see. We have info on this in the upgrading docs for 0.38 because the format of data on disk changes. However, people who have forks that play with the indexer, they might be impacted.